### ISSUE TITLE
Tank lets you go inside it after throwing rock

### ISSUE THREAD

**Issue by alexiscoutinho** [Tsuey/L4D2-Community-Update#63]
It seems like the Tank gets stuck in an animation after throwing the rock at someone else. It lets you go inside it and mess it up. https://youtu.be/vIoSk4UGJuU?t=2152

**Comment by Nesciuse** [comment]
This is not something that should ever get touched. Along with the minigun drawing the tank. This is a thing since l4d1 and melee fighting tank relies on this a lot. I would seriously get behind fixing a bunnyhop rather than this.

**Comment by alexiscoutinho** [comment]
:/ Sadge. So was this always intended then...? But even if melee fights "rely on it a lot", it is not necessary nor is the majority of the melee aspect. Kiting the Tank, which I think is the most used albeit highest skill technique, doesn't abuse it. At least visually this technique doesn't make the Tank look broken, it just looks noob.

**Comment by Nesciuse** [comment]
I am not sure if it's intended or why it happens. Or if it's an animation issue. It's just tank has locked aim or something like that little while after he throws rock. So making him throw a rock a running at him to the side is one of the ways to start a melee fight. I've seen it used many times in speedruns by just running around the tank. Imo it should be considered part of tank's behaviour and not a bug whatever is the reason it happens. This is one of those things where the only acceptable time period to change it was before the game released.

**Comment by alexiscoutinho** [comment]
I understand. I will just emphasize once again that it is just "one of the ways" to start a melee fight or run past it. There are others which don't look nearly as broken (https://youtu.be/A788Fd0IZ2w?t=125). Do you think the community would have the power to vote/decide on this behavior to justify a medium change years after release? Regardless, I would like to hear from some others before closing...

**Comment by Nesciuse** [comment]
The fight in the video you linked relies on this behaviour of tank having locked aim after throwing rock

**Comment by alexiscoutinho** [comment]
But the player waited for the Tank to come at him (reset state, no looking at someone else). I even set the timestamp up so that it showed the fight initiated much after the rock. The fight was initiated because he dodged the Tank's first punch, not because the Tank was looking at someone else and was chill with someone blasting it on its back for several seconds (enough time for more than 2 melee hits if the other guy had one). While the behaviors may be related, they can probably be dealt with separetely. It doesn't even look like the Tank is stuck in an animation in the second video, it just looks like he's recharging the next punch (which is enough time for only 2 hits [not broken]).

**Comment by Nesciuse** [comment]
It's the same behaviour. The punch can't be dodged like that without the locked aim after rock throw. Only additional thing in the first video is the player hid behind the tree so the tank changes target and starts throwing rock at someone else.

**Comment by alexiscoutinho** [comment]
Hmmm, didn't know about that. I will look more into it.

**Comment by alexiscoutinho** [comment]
@Nesciuse You are right. The only reliable way to initiate a Tank melee fight is after it throws a rock and the two videos use the exact same Tank behavior. So the mechanic can't be scrapped, but I will still propose to tweak it a bit. I suggest that the extra cooldown time gained after throwing a rock gets decreased the closer someone stands to the Tank. So there will still be, say, 5 seconds of grace period after the rock if everyone is far away. If there is someone right beside it after the rock, there would be just enough time for him to get away (dodge punch) or a minimum of 2 secs. This would allow the fight of the second video to happen because there would be just enough grace time for him dodge the punch, even if he charged sooner, and would prevent the Tank from chilling like in the first video as it would start attacking him soon after he touched it. In other words, the grace period could be cut short and immediately switch to the 2 hits per 2 seconds normal fight cycle/phase if someone charges the Tank.

**Comment by Nesciuse** [comment]
I really see no reason why you would want such changes. This is a thing that if changed would just annoy people that use this mechanic and people who don't use the mechanic wouldn't even know anything changed.

**Comment by alexiscoutinho** [comment]
It would mostly be an aesthetic and realism change, but would also make Tank fights a little bit longer as a team with guns might miss some shots if they have to stay a bit farther away from the Tank and because a melee team would need one, two or three extra 2 hits/2 secs cycles since they wouldn't be able to abuse the prolonged grace period. And I don't think annoying some melee players a little bit at first is relevant, by itself. A bunch of players got annoyed when many boost spots and unclipped props were patched and it doesn't matter. And there were no alternatives for all those people. The Tank behavior in those specific instances I mentioned could be considered as broken, unintended and unrealistic as those spots prior to TLS update. Furthermore, the hardcore melee group is small and they would quickly adapt too.

**Comment by alexiscoutinho** [comment]
The suggestion is invalidated, however, IF the reason behind the **fixed** 5 secs timer is to represent the **extra strain** on the Tank's arms due to throwing the rock ("physical" limitation), instead of some arbitrary limitation for balance. If Valve views it this way, then the most I can suggest is for the Tank to not charge survivors if it is at the beginning of the cooldown. Sure the Tank is *yolo* and dumb, but maybe it is too dumb if it waits the **super long** recharge kissing a machete (or several). The Tank could even be tweaked to avoid/run away from super nearby survivors if it's in the beginning of the recharge, say, in the first 2 seconds (3 to go). This wouldn't make a difference most of the time because survivors are usually not beside the Tank when it throws the rock. This change would allow the survivors to do melee fights, but would prevent them from abusing that bizarre Tank behavior. 

This fight/flight phase threshold could be controlled via a cvar, similar to #62 (check the advantages there). For example, if the threshold value is infinity, the Tank would never run away; if at 3.0 (good), the Tank would start charging super close survivors (remember the Tank doesn't change for medium distance+ survivors, it doesn't avoid them) when it has 3 secs to go to recharge the punch; if at 1.0 (bad!), the Tank would break the 2 hits/2 secs cycle since it would run away from the melee guy in the first second, initiating fights would probably be impossible aswell, unless the survivor corners the Tank or is using adrenaline. Most importantly the cvar would allow players and addons to test the change while retaining the classic behavior as default (at first).

**Comment by alexiscoutinho** [comment]
:(

No reply?!
